packages/neuron-wallet/src/services/wallets.ts

```diff
@@ -482,6 +488,299 @@ export default class WalletService {
+      throw new TransactionIsNotCommittedYet()
+    }
+    const content = cellStatus.cell.data!.content
+    const buf = Buffer.from(content.slice(2), 'hex')
+    const depositBlockNumber: bigint = buf.readBigUInt64LE()
+    const depositBlock = await core.rpc.getHeaderByNumber(depositBlockNumber)
+    const depositEpoch = this.parseEpoch(BigInt(depositBlock.epoch))
+    const depositCapacity: bigint = BigInt(cellStatus.cell.output.capacity)
+
+    const withdrawBlock = await core.rpc.getHeader(tx.txStatus.blockHash!)
+    const withdrawEpoch = this.parseEpoch(BigInt(withdrawBlock.epoch))
+
+    const withdrawFraction = withdrawEpoch.index * depositEpoch.length
+    const depositFraction = depositEpoch.index * withdrawEpoch.length
+    let depositedEpoches = withdrawEpoch.number - depositEpoch.number
+    if (withdrawFraction > depositFraction) {
+      depositedEpoches += BigInt(1)
+    }
+    const lockEpoches = (depositedEpoches + (DAO_LOCK_PERIOD_EPOCHS - BigInt(1))) / DAO_LOCK_PERIOD_EPOCHS * DAO_LOCK_PERIOD_EPOCHS
+    const minimalSinceEpochNumber = depositEpoch.number + lockEpoches
+    const minimalSinceEpochIndex = depositEpoch.index
+    const minimalSinceEpochLength = depositEpoch.length
+
+    const minimalSince = this.epochSince(minimalSinceEpochLength, minimalSinceEpochIndex, minimalSinceEpochNumber)
+
+    const outputCapacity: bigint = await this.calculateDaoMaximumWithdraw(depositOutPoint, withdrawBlock.hash)
+
+    const { codeHash, outPoint: secpOutPoint, hashType } = await LockUtils.systemScript()
+    const daoScriptInfo = await DaoUtils.daoScript()
+
+    const address = await AddressesService.nextUnusedAddress(walletID)
+    const blake160 = LockUtils.addressToBlake160(address!.address)
+
+    const output: Cell = {
+      capacity: (BigInt(outputCapacity) - finalFee).toString(),
+      lock: {
+        codeHash,
+        hashType,
+        args: blake160,
+      },
+      data: '0x'
+    }
+
+    const outputs: Cell[] = [output]
+
+    const previousOutput = TypeConvert.toOutput(cellStatus.cell.output)
+    const input: Input = {
+      previousOutput: withdrawingOutPoint,
+      since: minimalSince.toString(),
+      lock: previousOutput.lock,
+      lockHash: LockUtils.computeScriptHash(previousOutput.lock)
+    }
+
+    return {
+      version: '0',
+      cellDeps: [
+        {
+          outPoint: secpOutPoint,
+          depType: DepType.DepGroup,
+        },
+        {
+          outPoint: daoScriptInfo.outPoint,
+          depType: DepType.Code,
+        },
+      ],
+      headerDeps: [
+        depositBlock.hash,
+        withdrawBlock.hash
+      ],
+      inputs: [input],
+      outputs,
+      outputsData: outputs.map(o => o.data || '0x'),
+      witnesses: [],
+      witnessArgs: [{
+        lock: undefined,
+        inputType: '0x0000000000000000',
+        outputType: undefined,
+      }],
+      fee: finalFee.toString(),
+      interest: (BigInt(output.capacity) - depositCapacity).toString(),
```

Comment:
should the calculation of interest not include fee? interest calculation should be `outputCapacity - depositCapacity`?

Reply:
Not sure, but I think interest is `all - principal`, fee is another thing.

Reply:
OK, I see, it's my fault .. should exclude fee.

Reply:
Fixed.